files/build_templates/docker_image_ctl.j2

```diff
@@ -1,24 +1,49 @@
 #!/bin/bash
 
-# Obtain our platform and HWSKU as we will mount directories with these names in each docker
+# Obtain our platform as we will mount directories with these names in each docker
 PLATFORM=`sonic-cfggen -v platform`
+{%- if docker_container_name != "database" %}
+
+# Obtain our HWSKU as we will mount directories with these names in each docker
 HWSKU=`sonic-cfggen -m /etc/sonic/minigraph.xml -v minigraph_hwsku`
+{%- endif %}
 
 start() {
-    docker inspect --type container {{docker_container_name}} &>/dev/null
+    DOCKERCHECK=`docker inspect --type container {{docker_container_name}} 2>/dev/null`
     if [ "$?" -eq "0" ]; then
+{%- if docker_container_name == "database" %}
+        echo "Starting existing {{docker_container_name}} container"
         docker start {{docker_container_name}}
-    else
+        exit 0
+{%- else %}
+        DOCKERMOUNT=`echo $DOCKERCHECK | python -c "import sys, json, os; print os.path.basename([x for x in json.load(sys.stdin)[0]['Mounts'] if x['Destination'] == '/usr/share/sonic/hwsku'][0]['Source'])" 2>/dev/null`
+        if [ "$DOCKERMOUNT" == "$HWSKU" ]; then
```

Comment:
for database, the hwsku is not mounted, then if there is no dockermount let's do docker start then. In this case, you do no need to treat database specially.

Reply:
Change has been made according to our conversations.

The change here so far has been tested. If you are ok with it. I'll rebase on top of Taoyu's change and run another test before checking in.

Thanks,
Ying